src/test/scala/org/broadinstitute/hail/methods/SkatoSuite.scala

```diff
@@ -0,0 +1,387 @@
+                true
+              else {
+                println(s"group: $g")
+                println(s"PValue: plink=${p2._1} hail=${h2._1} ${p2._1 == h2._1}")
+                println(s"nMarkers: plink=${p2._2} hail=${h2._2} ${p2._2 == h2._2}")
+                println(s"nMarkersTested: plink=${p2._3} hail=${h2._3} ${p2._3 == h2._3}")
+                false
+              }
+            }.fold(true)(_ & _)
+            result
+          }
+        }
+      }
+
+    def compositeAnnotationGen = for (vds: VariantDataset <- VariantSampleMatrix.gen[Genotype](sc, VSMSubgen.random);
+                                      nGroups <- Gen.choose(1, 10);
+                                      genes <- Gen.buildableOfN[Array[String], String](nGroups, Gen.arbString);
+                                      phenotypes <- dichotomousPhenotype(vds.nSamples);
+                                      annotations <- Gen.buildableOfN[Array[Annotation], Annotation](vds.nVariants.toInt, genVariantAnnotation(genes));
+                                      annotType <- Gen.oneOf("Set", "Array")
+    ) yield (vds, annotations, genes, phenotypes, annotType)
+
+    property("handle splat") =
+      forAll(compositeAnnotationGen) {case (vds, annotations, genes, phenotypes, annotType) =>
+        val tmpOutput = tmpDir.createTempFile("splatTest")
+        val variantAnnotations = vds.variants.zip(sc.parallelize(annotations)).map{case (v, va) => (v, Annotation(va))}
+        val phenotypeMap = vds.sampleIds.zip(phenotypes.map{case p => Annotation(p.orNull)}).toMap
+
+        val groupKeyQueryString =
+          if (annotType == "Set")
+            s"let x = va.vep.transcript_consequences.map(csq => csq.gene_id).toSet in if (x.size == 0) NA: Set[String] else x"
+          else
+            s"let x = va.vep.transcript_consequences.map(csq => csq.gene_id) in if (x.length == 0) NA: Array[String] else x"
+
+        val splatResults = tmpOutput + ".splat.tsv"
+        val noSplatResults = tmpOutput + ".nosplat.tsv"
+        val configFile = tmpOutput + ".config"
+
+        var s = State(sc, sqlContext, vds
+          .annotateVariants(variantAnnotations, testVepSignature, List("vep"))
+          .annotateSamples(phenotypeMap, sampleSignature, List("pheno")))
+        s = SplitMulti.run(s, Array.empty[String])
+
+        val nNonMissingPheno = phenotypes.count(_.isDefined)
+        val nVariants = s.vds.nVariants
+
+        writeConfig(tmpOutput + ".config")
+
+        val cmdNoSplat = mutable.ArrayBuffer("--config", configFile ,
+          "-k", groupKeyQueryString,
+          "-y","sa.pheno.phenotype"
+        )
+        val cmdSplat = mutable.ArrayBuffer("--config", configFile,
+          "-k", groupKeyQueryString,
+          "-y","sa.pheno.phenotype"
+        )
+
+        cmdNoSplat += "-o"
+        cmdNoSplat += noSplatResults
+
+        cmdSplat += "-o"
+        cmdSplat += splatResults
+        cmdSplat += "--splat"
```

Comment:
Why make these mutable and modify them?  Why not put all the args in an Array constructor?